pkg/reconciler/common/images.go

```diff
@@ -138,18 +153,21 @@ func updateCachingImage(instance *servingv1alpha1.KnativeServing, u *unstructure
 }
 
 // updateImageSpec updates the image of a with a new registry and tag
-func updateImageSpec(image *caching.Image, registry *servingv1alpha1.Registry, log *zap.SugaredLogger) {
-	if newImage := getNewImage(registry, image.Name); newImage != "" {
+func updateImageSpec(image *caching.Image, registry *v1alpha1.Registry, log *zap.SugaredLogger) {
+	if newImage := getNewImage(registry, image.Name, ""); newImage != "" {
 		log.Debugf("Updating image from: %v, to: %v", image.Spec.Image, newImage)
 		image.Spec.Image = newImage
 	}
 	image.Spec.ImagePullSecrets = addImagePullSecrets(image.Spec.ImagePullSecrets, registry, log)
 	log.Debugw("Finished updating image", "image", image.GetName())
 }
 
-func getNewImage(registry *servingv1alpha1.Registry, containerName string) string {
-	if overrideImage := registry.Override[containerName]; overrideImage != "" {
-		return overrideImage
+func getNewImage(registry *v1alpha1.Registry, containerName, parent string) string {
+	if image, ok := registry.Override[parent+delimiter+containerName]; ok {
```

Comment:
This made the operator CR support `/containerName` as valid key when parent is empty. I know it does not seem harmful, but is this what you intentionally to enable?

Reply:
It wasn't intentional, but I'm not sure it's worth accounting for since `/` isn't a valid character in container names. At least, that's how I interpret this: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#path-segment-names